importer/importer.go

```diff
@@ -137,15 +140,22 @@ func (im *Importer) createTable(server servers.ClickhouseServer, user apikeys.AP
 		storagePolicy = server.GetStoragePolicy()
 	}
 
+	engine := "MergeTree"
+	clusterStmt := ""
+	if cluster := user.GetDBCluster(); cluster != "" {
+		engine = "ReplicatedMergeTree('/clickhouse/{cluster}/tables/{database}/{table}', '{replica}')"
```

Comment:
Where do the values for `{cluster}`, `{database}`, `{table}`, and `{replica}` come from?

Reply:
They are taken from the macro section in Clickhouse configuration file.

Reply:
Since a clickhouse physical server will have multiple clusters and databases, should we instead do this substitution using code instead of a macro file?

Reply:
The path is for replication-related metadata. It is essential for coordination between shards and replicas. Keeper uses it to track all parts of the data stored in the table, the log of changes, and the states of replicas. We should have Clickhouse handle this.

What are your concerns?

Reply:
My understanding is that clickhouse's config has 1 value for a macro. For we could have:

```
<macros>
  <database>db1</database>
</macros>
``` 

However, what if we have 2 databases, `db1` and `db2` on a host? How would clickhouse know which value of `database` to insert?

Reply:
@poundifdef I was not very clear in my first answer.

The cluster and replica placeholder are gotten from the macros. The table and database placeholder are from the database and table the operation is performed on. The values are replaced with the actual database and table names where the data will be stored. 

The `{database}` is the name of the ClickHouse database that contains the table. When you create a table, you either specify the database explicitly in the `CREATE TABLE` query or use the currently selected database if you're running commands interactively). The `{database}` in the Keeper path should match the name of the database you've used to create the table.
The `{table}` is the name of the table within the ClickHouse database. This is the specific table that you are making replicated across the cluster.
The `{table}` in the Keeper path is replaced with this unique table name.

For example, say we have a cluster  called `local` and a macros (macros are defined per node/replica):

```xml
<macros>
    <cluster>local</cluster>
    <replica>1</replica>
    <shard>1</shard>
</macros>
```
and the following query:

```sql
CREATE TABLE scratch.table
ON CLUSTER '{cluster}' (
    date Date,
    key Int,
    value String
)
ENGINE = ReplicatedMergeTree('/
clickhouse/{cluster}/tables/{shard}/{database}/{table}', '{replica}')
PARTITION BY toYYYYMM(date)
ORDER BY (date, key);
```

The Keeper path for the `ReplicatedMergeTree` above would be `ReplicatedMergeTree('/
clickhouse/local/tables/1/scratch/table', '1')` for the particular replica using the macros.

Ultimately, substituting the placeholder in our code will be the same as hardcoding the keeper path which will have negative effects especially when sharding.

Reply:
I see - I just looked it up, I didn't realize that `{table}` and `{database}` are not macros that are defined in the config. But here is the scenario:

- There is a physical clickhouse server on `1.1.1.1`
- That server has two clusters: `1` and ``

If my macro says the following: `<macros><cluster>1</cluster></macros>` then I would not be able to create a `ReplicatedMergeTree` on cluster `2`. Because the macro only has a value for `1`. In this example, we'd execute this query:

```
CREATE TABLE scratch.table
ON CLUSTER '1' (
    date Date,
    key Int,
    value String
)
ENGINE = ReplicatedMergeTree('/
clickhouse/1/tables/1/scratch/table', '1')
PARTITION BY toYYYYMM(date)
ORDER BY (date, key);
```
How would we make sure to handle this? 

The second case is if we rename a table. the clickhouse [docs](https://clickhouse.com/docs/en/engines/table-engines/mergetree-family/replication) say that the built-in macros will cause problems if a table is renamed:

> The two built-in substitutions {database} and {table} can be used, they expand into the table name and the database name respectively (unless these macros are defined in the macros section). So the zookeeper path can be specified as '/clickhouse/tables/{shard}/{database}/{table}'. Be careful with table renames when using these built-in substitutions. The path in ClickHouse Keeper cannot be changed, and when the table is renamed, the macros will expand into a different path, the table will refer to a path that does not exist in ClickHouse Keeper, and will go into read-only mode.